### ISSUE TITLE
Snowflake connector does not automatically surround field names in double-quotes

### ISSUE THREAD

**Issue by gs-scooter** [apache/superset#25769]
When using Snowflake as a data source for a chart, the chart will fail with a SQL compilation error. The only workaround is to go into the field's custom SQL and surround it with double-quotes. This bug exists in the preset.io product as well.

#### How to reproduce the bug

1. Launch Superset via a Docker container.
2. Add a Snowflake data source.
3. Create a new chart using the data source.
4. Drag fields onto the chart designer and create chart.
5. See compilation error referencing invalid identifier 'FIELD_NAME'.

### Expected results

No SQL compilation error. The Snowflake connector should work out of the box.

#### Screenshots

Error:

![image](https://github.com/apache/superset/assets/29981892/d0ff458e-a6c1-40fa-bc90-09ff50dcc5b8)

Workaround:

![image](https://github.com/apache/superset/assets/29981892/d9293f9e-75e2-4bef-b9ea-3c4f1ac44fe8)

### Environment

- browser type and version: Vivaldi 6.2.3105.58
- superset version: Version on latest [Docker image](https://hub.docker.com/r/apache/superset).
- any feature flags active: `snowflake-sqlalchemy` driver installed.

**Comment by sfirke** [comment]
@rusackas you are the sole person listed in the database rolodex for Snowflake - do you have any thoughts on this?

**Comment by rusackas** [comment]
I have access to a database to reproduce issues, but I don't use it on any real basis, or know much about _actually fixing it_. I think @villebro might use it too?

**Comment by wavenator** [comment]
We are using Snowflake too and having the same issue. Pretty frustrating no one is picking up this issue and fix it :/

**Comment by rusackas** [comment]
@wavenator please feel free, you have as much right as anyone to fix it, and if you're affected by it, you have all the more reason to. I understand the frustration, but there are an awful lot of issues, and never enough contributors.

**Comment by sfirke** [comment]
Possibly relevant: https://stackoverflow.com/a/68311509

**Comment by mistercrunch** [comment]
Yes it does seem related to case sensitivity. It'd help to get more information on the schema's case and the SQL generated. Also relevant -> https://community.snowflake.com/s/article/How-to-ignore-the-case-of-quoted-identifiers-on-Snowflake#:~:text=Quoted%20identifiers%20are%20identifiers%20such,you%20will%20get%20an%20error.

**Comment by rusackas** [comment]
Tempted to close this as stale, but is anyone still facing this in 4.1.1 or 5.0.0rc1?

**Comment by wavenator** [comment]
Yep still experiencing this issue

**Comment by rusackas** [comment]
Did the two links/references above provide any help here? I don't have a Snowflake instance to test those suggestions on... maybe @sadpandajoe has one?

**Comment by sadpandajoe** [comment]
Is this something that sqlglot would normally handle @betodealmeida ?

**Comment by sadpandajoe** [comment]
@rusackas not able to really repro this unless if I use a temporal column, I have number columns that are all lower case and others that are mixed and upper case and they seem to work fine for the x-axis without having to use custom sql.

**Comment by rusackas** [comment]
Checking in — @sadpandajoe was unable to reproduce in June 2025 unless using temporal columns specifically. Is this still affecting anyone on Superset 4.1.x or 5.0? If we can't get a reproducible case, this may be closed.

**Comment by rusackas** [comment]
We've been unable to reproduce this on recent versions except in a narrow temporal-column case, and we haven't gotten a minimal reproduction. Closing as not reproducible — please reopen with a concrete dataset/chart definition on a current release (6.x.x) if you're still hitting this.
